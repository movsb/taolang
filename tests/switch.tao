function main() {
    switch 1 {
    case 1:
        switch 2 {
        case 1,2,3:
            println(__case);
            break;
            println("won't go here");
        default:
            println("2d");
        }
        println("11");
    default:
        println("?");
    }
    {
        let a = 1;
        switch a {
        case +a:
            println("+a");
        case a > 0:
            println("a > 0");
        case 1:
            println("1");
        }
    }
    println("end main");
}
