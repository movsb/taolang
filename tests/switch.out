2
11
+a
end main
